### ISSUE TITLE
UI Glitches

### ISSUE THREAD

**Issue by tlhackque** [sqlitebrowser/sqlitebrowser#1468]
Details for the issue
--------------------

#### What did you do?
Put a slow filter on a large table in the browser tab.

While it was running:

Switched to execute SQL tab and ran a SELECT on the same table

#### What did you expect to see?
Result on SQL tab.
Nothing on the cancelled browser tab.

#### What did you see instead?
1) The Browse tab turns pink and EVERY cell reports "loading".  It also does this when scrolling large distances on a big selection.  I think one "loading" in the pane would be less cluttered and would require fewer attention units to read.

2) The SQL tab produced a message box that said "the database is busy loading in another window, do you want to cancel?".  I said "yes", and got my result.  That's all fine.

3) When I switched back to the browser tab, the pane was blank - and the bottom left status said "determining row count".  After a (fairly long) while, the pane refreshed.

This is unexpected.  I said to "cancel" the update - but it reanimated much later.  There was no pink "loading" display, nor was there a "refreshing" message or busy spinner.

I'd prefer that once cancelled, it would stay cancelled until I provide a new filter or hit refresh.
But if it's going to reanimate, it should provide feedback/status.

Along the same lines, character by character queries are nice on indexed columns/small databases, but very expensive on large ones.  Perhaps a "delay filter" button, that when depressed, doesn't send the SQL, and when released, does?  That would give the current behavior by default (button out), but make larger applications scale better.

If I hide a number of columns, and discover that I want one back, I have to unhide ALL columns, then select and hide again.  It would be nice to have a way to selectively un-hide (again, this is an issue with large/slow queries.)  I run into this frequently - I hide a lot of columns so there's space to see the rightmost column - then need one or two to narrow the selection.  And the ones I hide are discontiguous.

Finally, it would be nice to be able to reorder (drag?) columns so that they appear in a different order from the table declaration.  That would let me put the columns applicable to a task together and save scrolling (and hiding/unhiding).

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [ x] Windows: ( _version:_10 pro ___ )
- [ ] Linux: ( _distro:_ ___ )
- [ ] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?

- [ ] 3.10.1
- [ ] 3.10.0
- [ ] 3.9.1
- [ ] Other: ___7-Jul

#### Did you also

- [ x] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [ ] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by tlhackque** [comment]
Oh, and one more:  When I click to drag-select in order to hide columns, it often initiates an undesired sort on the first column clicked.  Again, an issue on large datasets.  And when I'd previously sorted on a now-hidden column...

**Comment by mgrojo** [comment]
> Along the same lines, character by character queries are nice on indexed columns/small databases, but very expensive on large ones. Perhaps a "delay filter" button, that when depressed, doesn't send the SQL, and when released, does? That would give the current behavior by default (button out), but make larger applications scale better.

Maybe one more of those "nice hidden features". You can try with high values in the setting for **Data Browser > Filters > Delay time (ms)**. Ideally it should be possible to completely disable the automatic filtering, so it only filters after <kbd>Return</kbd> is pressed.

**Comment by tlhackque** [comment]
>  Ideally it should be possible to completely disable the automatic filtering, so it only filters after Return is pressed.

Agree.  There's lots of space if the value dropdown is made narrower.  Perhaps "Update while typing" vs, "Update after &lt;return&gt;", where the delay time is only enabled by selecting the former?

Might consider a lower bound on the value - and increasing the default.  I've seen statistics indicating that the average person types at about 40 "words" per minute, which is about 200 characters per minute.  This would be a delay of about 300 msec.  Even that seems aggressive here - this is not like typing natural language.  Perhaps 1 sec would be a better default?

"Delay time" isn't very obvious - not clear what the best alternative is: "Update delay time"? "Typein delay time"?  

Note that the "What's this" (?) on Edit->Preferences->Data Browser is dead for the entire tab.  Actually, it's dead for all the tabs!  Most of the items could use a sentence or two of explanation.

"?" should be active for "preferences" (or removed until it works).
